flybywiresim/a32nx
# feat: Manual ILS Tuning by Ident, Course

Fixes #2568 
Fixes #5788

## Summary of Changes

ILS' can now be tuned by ident on the rad nav page. ILS course can now be entered on the RAD NAV page. The course dagger on the ND and PFD is sourced from the correct data depending on entry, and availability of database or radio info. The `RWY/LS MISMATCH` scratchpad message is implemented, and is generated when the ILS/LOC manually tuned doesn't match the database one for the runway, or the entered course does not match the database.

The new facility search API introduced in SU6 returns ILS in the VOR search results, making this possible. This PR switches the data manager from the old FS9GPS search API to the new one for all search types, yielding a substantial improvement in performance for all database searches for idents.

## Screenshots (if necessary)


![image](https://user-images.githubusercontent.com/9995998/138021530-bb61c4d5-3872-46f2-967b-6d86e7eac6f0.png)

## References





Honeywell manual.

## Additional context



Discord username (if different from GitHub):

## Testing instructions

- Ensure ILS auto-tuning works properly (departure runway when at origin airport, approach frequency when within 300 NM of destination during cruise phase or later).
- Check manual ILS tuning by frequency, including clearing... should revert to autotuned ILS when available.
- Check manual ILS tuning by identifier, including clearing... should revert to autotuned ILS when available.
- Tune the wrong ILS for the runway selected, and ensure `RWY/LS MISMATCH` is generated.
- Check VOR and NDB tuning by frequency and ident
- Enter waypoints etc. into the flightplan and/or PROG BRG/DIST from the scratchpad
- Enter correct course on the RAD NAV page, and check no scratchpad message is generated
- Enter incorrect course (more than 3 degrees off proper course) and check that `RWY/LS MISMATCH` is generated.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
